File: java/src/main/java/ch/planval/datavisualizer/model/CountyData.java
package ch.planval.datavisualizer.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Data
@Entity
@Table(name = "calculated_data_view")
@AllArgsConstructor
@RequiredArgsConstructor
public class CountyData {
    @Id
    private Long id;

    private String name;
    private String comment;
    private String unit;
    private String topic;
    private String dimension;
    private String county;
    private int zip;
    private double raw_value;
    private int calculated_value;
}

File: ui/sustainability-ui/src/app/models/countyData.ts
export class CountyData {
    id: string;
    county: string;
    zip: number;
    comment: string;
    unit: string;
    topic: string;
    dimension: string;
    raw_value: number;
    calculated_value: number;
    name: string;

    constructor(id: string, county: string, zip: number, comment: string, unit: string, topic: string, dimension: string, raw_value: number, calculated_value: number, name: string) {
        this.id = id;
        this.county = county;
        this.zip = zip;
        this.comment = comment;
        this.unit = unit;
        this.topic = topic;
        this.dimension = dimension;
        this.raw_value = raw_value;
        this.calculated_value = calculated_value;
        this.name = name;
    }
}
